### PR TITLE
AddCO to ggroup-to-splunk

### LINKED ISSUE #23: Setup instance for monitoring cabinetoffice.gov.uk domain
**Note: The CO domain does not have a hyphen - it's `cabinetoffice.gov.uk`**

- [x] Update these instructions for creating appropriate credentials [docs](https://docs.google.com/document/d/1bOFh135Xd9wtW7tw7TmMJ09zS8cQZdwZk_90hdn0w-c/edit?usp=drive_web&ouid=117809841920698986644) - possibly with Owain Hopkins who created the GDS ones. 
- [x] Get CO domain admin (maybe Noor Shimul) to create us some GSuite creds for the scanner
- [x] Stick creds in SSM `/cabinetoffice.gov.uk/google_data_to_splunk/credentials`
- [x] Refactor to enable setup with different SSM parameter prefixes for different domains
   - Or deploy different domains to different AWS accounts.
- [x] Deploy a new instance of this for with the CO creds
- [x] get new data into splunk

### PR DESCRIPTION
**Pull request by alice-carr**
adding a couple of things that were missed last pr, including:
- specifying domain in terraform and passing it through to the lambda, 
- adding the co prefix to the subject email
- adding error handling to the actual script as that was breaking occasionally and sending no data through at all

relates #23

### CODE REVIEW

File: terraform/deployments/dev/main.tf [hunk 1]
```diff
@@ -6,4 +6,5 @@ module "ggroup-to-splunk-dev" {
   lambda_memory       = 256
   lambda_role_name    = "ggroup_lambda_exec_role"
   lambda_timeout      = 900
+  domain              = "digital.cabinet-office.gov.uk"
```
Comment (danjoneslf) [inline terraform/deployments/dev/main.tf:9]: Should we change this dev one to the `cabinet-office.co.uk` domain? Not on this PR but as a tech-debt issue for later.

**Review by danjoneslf — APPROVED** [review]
Nice. Fine by me.

File: lambda/script.py [hunk 2]
```diff
@@ -80,25 +81,30 @@ def build_group_dict(api: str, api_version: str, scope: str) -> Dict[str, str]:
...
+                    group_ids[group_names] = group_id
+
+                if "nextPageToken" not in groups:
+                    break
+                else:
+                    nextPageToken = groups["nextPageToken"]
+
+        except HttpError as e:
+            print(f"Http error getting group ids: {e}")
+            continue
```
Comment (0atman) [inline lambda/script.py]: Can these really be `continue`d?

Comment (alice-carr) [reply lambda/script.py]: you're right, these `continue`s may be doing absolutely nothing now I think of it - I just wanted to ensure it went to the next round of the for loop

File: terraform/deployments/dev/main.tf (hunk 1 above)
Comment (alice-carr) [reply terraform/deployments/dev/main.tf:9]: yeah great idea - I'll write a ticket for that now

**Pull request merged**